File: back/KaemeWebApi/Dominio/Services/PedidoService.cs
using Dominio.Interface;
using Dominio.Models;
using Shared.RepositoryShared;

namespace Dominio.Services
{
    public class PedidoService
    {
        private IPedidoRepository _pedidoRepository;
        private IPecaRepository _pecaRepository;

        public PedidoService(IPedidoRepository pedidoRepository, IPecaRepository pecaRepository)
        {
            _pedidoRepository = pedidoRepository;
            _pecaRepository = pecaRepository;
        }

        public RepositoryResult Create(PedidoCreate pedido)
        {
            try
            {
                var result = _pedidoRepository.Create(pedido);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public async Task<RepositoryResult> Delete(PedidoInformation pedido)
        {
            try
            {
                var result = await _pedidoRepository.Delete(pedido);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }        

        public RepositoryResult AtualizarPedidoCobranca(PedidoCobranca pedidoCobranca, bool baixar)
        {
            try
            {
                var result = _pedidoRepository.AtualizarPedidoCobranca(pedidoCobranca, baixar);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult AtualizarStatusPedido(int num_pedido, int status_pedido_id)
        {
            try
            {
                var result = _pedidoRepository.AtualizarStatusPedido(num_pedido, status_pedido_id);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult GetPedidoBySearch(PedidoSearch pedidoSearch)
        {
            try
            {
                var result = _pedidoRepository.GetPedidoBySearch(pedidoSearch);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public async Task<RepositoryResult> GetPedidoByNumPedido(int num_pedido)
        {
            try
            {
                PedidoCreate pedidoCreate = new PedidoCreate();

                var pedido = await _pedidoRepository.GetPedidoByNumPedido(num_pedido);
                var pedidoPeca = await _pedidoRepository.GetPedidoPecaByNumPedido(num_pedido);
                var pedidoCobranca = await _pedidoRepository.GetPedidoCobrancaByNumPedido(num_pedido);
                var pecasReturn = await _pecaRepository.GetPecasByNumPedido(num_pedido);


                pedidoCreate.Pedido = pedido;
                pedidoCreate.Pecas = pedidoPeca;
                pedidoCreate.Pedido_Cobranca = pedidoCobranca;
                pedidoCreate.PecasReturn = pecasReturn;


                return RepositoryResult.AddDapper(pedidoCreate);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult GetTop100Pedidos()
        {
            try
            {
                var result = _pedidoRepository.GetTop100Pedidos();
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public RepositoryResult GetPedidoCliente(int num_pedido, string cpf)
        {
            try
            {
                var result = _pedidoRepository.GetPedidoCliente(num_pedido, cpf);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }

        public List<PedidoInformation> GetHistoricoPedido(PedidoSearch pedidoSearch)
        {
            var result = _pedidoRepository.GetHistoricoPedido(pedidoSearch);
            return result;
        }

        public RepositoryResult ConfirmarRecebimento(int num_pedido)
        {
            try
            {
                var result = _pedidoRepository.ConfirmarRecebimento(num_pedido);
                return RepositoryResult.AddDapper(result);
            }
            catch (Exception e)
            {
                return RepositoryResult.AddException(e);
            }
        }        


    }
}

File: front/kaemeweb/src/app/modules/shared/services/pedido.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, tap } from 'rxjs';
import { Cliente } from '../models/cliente';
import { PedidoCreate } from '../models/PedidoModels/pedido-create';
import { PedidoSearch } from '../models/PedidoModels/pedido-search';
import { PedidoInformation } from '../models/PedidoModels/pedido-information';
import { PedidoCobranca } from '../models/PedidoModels/pedido-cobranca';
import { Pedido } from '../models/PedidoModels/pedido';

@Injectable({
  providedIn: 'root'
})
export class PedidoService {

  constructor(private httpClient: HttpClient) { }

  create(pedidoCreate: PedidoCreate): Observable<Pedido> {
    return this.httpClient.post<Pedido>('https://localhost:7072' + '/pedido/Create', pedidoCreate);
  }

  delete(pedido: PedidoInformation): Observable<boolean> {
    return this.httpClient.post<boolean>('https://localhost:7072' + '/pedido/Delete', pedido);
  }

  getPedidoBySearch(pedidoSearch: PedidoSearch): Observable<PedidoInformation[]> {
    return this.httpClient.post<PedidoInformation[]>('https://localhost:7072' + '/pedido/GetPedidoBySearch', pedidoSearch);
  }

  getTop100Pedidos(): Observable<PedidoInformation[]> {
    return this.httpClient.get<PedidoInformation[]>('https://localhost:7072' + '/pedido/GetTop100Pedidos');
  }

  getPedidoByNumPedido(num_pedido: number): Observable<PedidoCreate> {
    let params = new HttpParams()
    params = params.append('num_pedido', num_pedido)
    return this.httpClient.get<PedidoCreate>('https://localhost:7072' + '/pedido/GetPedidoByNumPedido', {params});
  }

  atualizarPedidoCobranca(pedidoCobranca: PedidoCobranca, baixar: boolean): Observable<boolean> {
    let params = new HttpParams()
    params = params.append('baixar', baixar)
    return this.httpClient.post<boolean>('https://localhost:7072' + '/pedido/AtualizarPedidoCobranca', pedidoCobranca, {params});
  }

  atualizarStatusPedido(num_pedido: number, status_pedido_id: number): Observable<boolean> {
    let params = new HttpParams()
    params = params.append('num_pedido', num_pedido)
    params = params.append('status_pedido_id', status_pedido_id)

    return this.httpClient.get<boolean>('https://localhost:7072' + '/pedido/AtualizarStatusPedido?num_pedido=' +num_pedido + '&status_pedido_id='+status_pedido_id);
  }

  getPedidoCliente(num_pedido: number, cpf: string): Observable<PedidoInformation[]> {
    let params = new HttpParams()
    params = params.append('num_pedido', num_pedido)
    params = params.append('cpf', cpf)
    return this.httpClient.get<PedidoInformation[]>('https://localhost:7072' + '/pedido/GetPedidoCliente', {params});
  }

  confirmarRecebimento(num_pedido: number): Observable<boolean> {
    let params = new HttpParams()
    params = params.append('num_pedido', num_pedido)

    return this.httpClient.get<boolean>('https://localhost:7072' + '/pedido/ConfirmarRecebimento?num_pedido=' +num_pedido);
  }

}
